packages/amplify-graphql-model-transformer/src/resources/dynamo-model-resource-generator.ts

```diff
@@ -0,0 +1,301 @@
+    cfnTable.sseSpecification = {
+      sseEnabled: cdk.Fn.conditionIf(useSSE.logicalId, true, false),
+    };
+
+    const streamArnOutputId = `GetAtt${ModelResourceIDs.ModelTableStreamArn(def!.name.value)}`;
+    // eslint-disable-next-line no-new
+    new cdk.CfnOutput(stack, streamArnOutputId, {
+      value: cdk.Fn.getAtt(tableLogicalName, 'StreamArn').toString(),
+      description: 'Your DynamoDB table StreamArn.',
+      exportName: cdk.Fn.join(':', [context.api.apiId, 'GetAtt', tableLogicalName, 'StreamArn']),
+    });
+
+    const tableNameOutputId = `GetAtt${tableLogicalName}Name`;
+    // eslint-disable-next-line no-new
+    new cdk.CfnOutput(stack, tableNameOutputId, {
+      value: cdk.Fn.ref(tableLogicalName),
+      description: 'Your DynamoDB table name.',
+      exportName: cdk.Fn.join(':', [context.api.apiId, 'GetAtt', tableLogicalName, 'Name']),
+    });
+
+    const role = this.createIAMRole(context, def, stack, tableName);
+    const tableDataSourceLogicalName = `${def!.name.value}Table`;
+    this.createModelTableDataSource(def, context, table, stack, role, tableDataSourceLogicalName);
+  }
+
+  private createModelTableDataSource(
+    def: ObjectTypeDefinitionNode,
+    context: TransformerContextProvider,
+    table: Table,
+    stack: cdk.Stack,
+    role: iam.Role,
+    dataSourceLogicalName: string,
+  ): void {
+    const datasourceRoleLogicalID = ModelResourceIDs.ModelTableDataSourceID(def!.name.value);
+    const dataSource = context.api.host.addDynamoDbDataSource(
+      datasourceRoleLogicalID,
+      table,
+      { name: dataSourceLogicalName, serviceRole: role },
+      stack,
+    );
+
+    const cfnDataSource = dataSource.node.defaultChild as CfnDataSource;
+    cfnDataSource.addDependsOn(role.node.defaultChild as CfnRole);
+
+    if (context.isProjectUsingDataStore()) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const datasourceDynamoDb = cfnDataSource.dynamoDbConfig as any;
+      datasourceDynamoDb.deltaSyncConfig = {
+        deltaSyncTableName: context.resourceHelper.generateTableName(SyncResourceIDs.syncTableName),
+        deltaSyncTableTtl: '30',
+        baseTableTtl: '43200',
+      };
+      datasourceDynamoDb.versioned = true;
+    }
+
+    const datasourceOutputId = `GetAtt${datasourceRoleLogicalID}Name`;
+    // eslint-disable-next-line no-new
+    new cdk.CfnOutput(stack, datasourceOutputId, {
+      value: dataSource.ds.attrName,
+      description: 'Your model DataSource name.',
+      exportName: cdk.Fn.join(':', [context.api.apiId, 'GetAtt', datasourceRoleLogicalID, 'Name']),
+    });
+
+    // add the data source
+    context.dataSources.add(def!, dataSource);
+    this.datasourceMap[def!.name.value] = dataSource;
+  }
+
+  /**
+   * createIAMRole
+   */
+  createIAMRole = (context: TransformerContextProvider, def: ObjectTypeDefinitionNode, stack: cdk.Stack, tableName: string): iam.Role => {
```

Comment:
Should this be 'private'? Also, use the same function style for the class methods.

Reply:
it's public for the sake of passing through calls from the model transformer plugin if any are still made, those methods still have to be there to not break the API. Although we may be able to remove them since RDS should be a major version bump